src/main/java/org/lsc/jndi/JndiServices.java

```diff
@@ -903,6 +929,15 @@ private boolean doApply(final JndiModifications jm) throws CommunicationExceptio
 			
 			return false;
 		}
+		finally {
+			// clear requestControls for future use of the JNDI context
+			try {
+				ctx.setRequestControls(null);
```

Comment:
Do you think this configuration of the LdapContext is thread-safe? It seems ctx is shared by all threads using JndiServices, no?

Reply:
you're right, I changed the way we use this control to derive the context for updates so the initial/shared context is not impacted by the control.